hvplot/util.py

```diff
@@ -320,10 +320,7 @@ def is_dask(data):
     return isinstance(data, (dd.DataFrame, dd.Series))
 
 def is_intake(data):
-    if not check_library(data, 'intake'):
-        return False
-    from intake.source.base import DataSource
-    return isinstance(data, DataSource)
+    return any(check_library(cls, 'intake') for cls in type(data).mro())
```

Comment:
```suggestion
    if "intake" not in sys.modules:
        return False
    from intake.source.base import DataSource
    return isinstance(data, DataSource)
```

I think we need to have the DataSource check, for an edge cases like this will return True:
![image](https://user-images.githubusercontent.com/19758978/228778065-1c942c1a-4eeb-4f93-b427-19f9b964c7f1.png)

Reply:
You could still have both, then - replace the current library check with the modified one, and then import as before.

Reply:
No, it is currently failing for subclasses of DataSource which are defined outside of the intake package.